src/Servers/IIS/IIS/src/Core/IISHttpContextOfT.cs

```diff
@@ -47,7 +53,7 @@ public override async Task<bool> ProcessRequestAsync()
             {
                 _streams.Stop();
 
-                if (!HasResponseStarted && _applicationException == null && _onStarting != null)
+                if (!HasResponseStarted && _requestRejectedException == null && _applicationException == null && _onStarting != null)
```

Comment:
Kestrel doesn't check _requestRejectedException before calling FireOnStarting post _application.ProcessRequestAsync(). Should it?

Reply:
I think it should, but it would be a "breaking" behavioral change. I'm going to remove the check for IIS for now and we can discuss adding it to both later.